Watchman.Discord/Areas/Protection/Services/UserMessagesCountService.cs

```diff
@@ -21,24 +22,45 @@ public UserMessagesCountService(IQueryBus queryBus)
 
         public int CountMessages(Contexts contexts)
         {
-            if (_serverMessagesQuantity == null || _lastUpdated.AddHours(12) < DateTime.UtcNow)
+            if (ShouldReloadCache())
             {
-                _lastUpdated = DateTime.UtcNow;
+                ReloadCache();
+            }
+
+            return GetOneUserFromOneServerCount(contexts.User.Id, contexts.Server.Id);
+        }
 
-                var query = new GetMessagesQuery();
-                var messages = _queryBus.Execute(query).Messages.ToList();
+        private bool ShouldReloadCache()
+        {
+            return _everyServersMessagesQuantity == null || _lastUpdated.AddHours(12) < DateTime.UtcNow;
+        }
 
-                var serversMessages = messages
-                    .GroupBy(x => x.Server.Id)
-                    .ToList();
+        private void ReloadCache()
+        {
+            _lastUpdated = DateTime.UtcNow;
 
-                this._serverMessagesQuantity = serversMessages
-                    .Select(x => new ServerMessagesCount(x.ToList(), x.Key))
-                    .ToList();
-            }
+            var getAllMessages = new GetMessagesQuery(0);
+            var messages = _queryBus.Execute(getAllMessages).Messages.ToList();
```

Comment:
tak właściwie to co ta klasa tak właściwie to co ta klasa ma robić?
liczyć dane do AntiSpamu czy do statystyk?

jeśli do antispamu to czemu używa bazy, to mało optymalne, i tak trzymamy dane w pamięci 

jeśli to ma być generowanie statystyk na przyszłość, to czemu per użytkownik?

Reply:
Dla antispam, z bazy pobiera raz na 12h, a co wiadomość pobiera z RAMu